### PR TITLE
Default follow-up requests

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds default follow-up requests.

### CODE REVIEW

File: skyportal/facility_apis/sedmv2.py [hunk 1]
```diff
@@ -117,12 +119,13 @@ def submit(request, session):
 
         session.add(transaction)
 
-        flow = Flow()
-        flow.push(
-            '*',
-            'skyportal/REFRESH_SOURCE',
-            payload={'obj_key': request.obj.internal_key},
-        )
+        if 'refresh_source' in kwargs and kwargs['refresh_source']:
```
Comment (guynir42) [inline skyportal/facility_apis/sedmv2.py:122]:
```suggestion
        if kwargs.get('refresh_source', False):
```

File: skyportal/handlers/api/followup_request.py [hunk 2]
```diff
@@ -1375,3 +1400,197 @@ async def put(self):
...
 
             return self.success()
+
+
+class DefaultFollowupRequestHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Create default follow-up request requests.
```
Comment (guynir42) [inline skyportal/handlers/api/followup_request.py:1410]:
```suggestion
        description: Create default follow-up request.
```

File: skyportal/handlers/api/followup_request.py [hunk 3]
```diff
@@ -1375,3 +1400,197 @@ async def put(self):
...
+        with self.Session() as session:
+
+            stmt = DefaultFollowupRequest.select(session.user_or_token).where(
+                DefaultFollowupRequest.id == default_followup_request_id
+            )
+            default_followup_request = session.scalars(stmt).first()
+
+            if default_followup_request is None:
+                return self.error(
+                    'Default follow-up request with ID {default_followup_request_id} is not available.'
```
Comment (guynir42) [inline skyportal/handlers/api/followup_request.py:1590]:
```suggestion
                    f'Default follow-up request with ID {default_followup_request_id} is not available.'
```

File: skyportal/handlers/api/followup_request.py [hunk 4]
```diff
@@ -1375,3 +1400,197 @@ async def put(self):
...
+            content:
+              application/json:
+                schema: Success
+        """
+
+        with self.Session() as session:
+
+            stmt = DefaultFollowupRequest.select(session.user_or_token).where(
+                DefaultFollowupRequest.id == default_followup_request_id
+            )
```
Comment (guynir42) [inline skyportal/handlers/api/followup_request.py:1585]: I think you should also specify that you need permissions to delete when making the select statement.

File: skyportal/models/followup_request.py [hunk 5]
```diff
@@ -203,3 +279,45 @@ def set_time(self, altitude=30 * u.degree):
...
+
+    # Add front-end user notifications
+    @event.listens_for(inspect(target).session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+
+        from skyportal.handlers.api.followup_request import post_followup_request
+
+        comp_function = getattr(operator, 'eq')
+        target_data = target.to_dict()
+        print(target_data)
```
Comment (guynir42) [inline skyportal/models/followup_request.py:295]:
```suggestion
```

File: skyportal/models/followup_request.py [hunk 6]
```diff
@@ -203,3 +279,45 @@ def set_time(self, altitude=30 * u.degree):
     AccessibleIfUserMatches('followuprequest.requester')
     & FollowupRequestTargetGroup.read
 )
+
+
+@event.listens_for(Classification, 'after_insert')
```
Comment (guynir42) [inline skyportal/models/followup_request.py:284]: What do classifications have to do with this?

File: static/js/components/NewDefaultFollowupRequest.jsx [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    };
+    await dispatch(
+      defaultFollowupRequestsActions.submitDefaultFollowupRequest(json)
+    );
+  };
+
+  console.log("instrumentList", instrumentList);
+  console.log("instrumentFormParams", instrumentFormParams);
+  console.log("allocationLookUp", allocationLookUp);
+  console.log("selectedAllocationId", selectedAllocationId);
```
Comment (guynir42) [inline static/js/components/NewDefaultFollowupRequest.jsx:158]:
```suggestion
```

**Review by guynir42 — APPROVED** [review]
Overall it looks good. I have a few comments, mostly minor ones.

File: skyportal/handlers/api/followup_request.py (hunk 4 above)
Comment (mcoughlin) [reply skyportal/handlers/api/followup_request.py:1585]: Good point.

File: skyportal/models/followup_request.py (hunk 6 above)
Comment (mcoughlin) [reply skyportal/models/followup_request.py:284]: In this first version, I am setting triggering criteria based on classification.

**Pull request merged**